### DISCUSSION TITLE
How might I apply the visx XYchart scale to SplitLinePath

### DISCUSSION THREAD

**Discussion by alicewuu** [airbnb/visx#1543] [Q&A]
Hi,

I am trying to scale the SplitLinePath to align with a LineSeries from the XYChart, I found that LineSeries is scaled with the [getScaledValueFactory](https://github.com/airbnb/visx/blob/c8dda524238190b59b8385dbedd6ee53f6779352/packages/visx-xychart/src/utils/getScaledValueFactory.ts#L7) and am trying to imitate that behavior with SplitLinePath. I have tried

`                <g transform="translate(0)">
                  <SplitLinePath
                    sampleRate={1}
                    styles={...}
                    segments={segments}
                    x={d =>
                      scaleLinear({
                        range: [0, chartWidth],
                        domain: [0, data.length],
                      })(d.x)
                    }
                    y={d =>
                      scaleLinear({
                        range: [chartHeight, 0],
                        domain: [
                          0,
                          Math.max(data.map(d => d.y || 1)),
                        ],
                      })(d.y)
                    }
                    segmentation={'x'}
                  />
                </g>
`
but it does not look quite right

<img width="545" alt="Screen Shot 2022-07-21 at 11 32 18 AM" src="https://user-images.githubusercontent.com/35696375/180287781-d9e50095-28cb-4a60-9f87-2a33bab7b41d.png">

Here is my LineSeries code for reference:
`
              <XYChart
                theme={customTheme}
                height={chartHeight}
                width={chartWidth}
                xScale={{ type: 'linear' }}
                yScale={{ type: 'linear' }}
                margin={{
                  top: chartMargin,
                  left: chartMargin,
                  bottom: chartMargin,
                  right: chartMargin,
                }}
              >
                <LineSeries
                  dataKey="LineSeries"
                  data={d}
                  {...accessors}
                  strokeWidth={2}
                />
              </XYChart>
`

**Comment by williaster** [comment]
Hi @alicewuu 👋  You can easily access the `XYChart` from a child component using its [`DataContext`](https://github.com/airbnb/visx/blob/master/packages/visx-xychart/src/types/data.ts#L34), it would look something like this:

```tsx
import React, { useContext } from "react";
import { DataContext, LineSeries, XYChart, ... } from "@visx/xychart";

function MyChart() {
  return (
    <XYChart {...props}>
      <LineSeries {...lineProps} />
      <CustomSeries />
    </XYChart>
  );
}

function CustomSeries() {
  const { xScale, yScale, ... } = useContext(DataContext);
  // use scales with `SplitLinePath`, note they may be `null` on first render
}
```

I think that's probably all you need to overlay a `SplitLinePath`? If you want the `XYChart` scales to account for data in your `CustomSeries` component, it's a bit more complicated but totally possible. This is the essence of how all the other series work, they simply invoke the `registerData` function with [`key/data/xAccessor/yAccessor/...`](https://github.com/airbnb/visx/blob/master/packages/visx-xychart/src/types/data.ts#L15) on mount and the context computes the extent of x/y data using that info across all series which register data.
